x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_data_client.ts

```diff
@@ -25,7 +27,10 @@ interface AssetCriticalityUpsert {
 
 type AssetCriticalityIdParts = Pick<AssetCriticalityUpsert, 'idField' | 'idValue'>;
 
+const MAX_CRITICALITY_RESPONSE_SIZE = 10000; // TODO what's the right value here?
```

Comment:
The TL;DR this value should probably be synchronized with the `page_size` configuration of the relevant risk scoring request. As it stands, there is potential (albeit unlikely) to not count asset criticality if users are modifying their task configuration:

1. User changes their task config from a `page_size` of 1000 to > 10000
2. During task execution, we receive > 10000 entities as part of a "page" to process
3. When we then go to retrieve asset criticality for those > 10k entities, only up to 10k will be returned

Again, this is an unlikely situation as the user needs both a) a modified task config and b) a lot of asset criticality records, but I want to call it out here to see if we think it should be addressed now or put in the backlog for later.

Reply:
Yes, it probably should be something like Math.max(10000, page_size)

Reply:
@nkhristinin I ended up compromising a bit here: we specify a `size` when invoking from our scoring function, which is based on the number of identifiers specified. Additionally, I've updated the `search` method to have both a default size (1000) and a maximum size (100_000) so that we don't accidentally try to throw a million of those records into memory. d64f772583faf691a827d329e5476b3fbab3cc04